tests/vcs/git/test_backend.py

```diff
@@ -41,3 +47,25 @@ def test_invalid_revision_sha_max_len() -> None:
 def test_get_name_from_source_url(url: str) -> None:
     name = Git.get_name_from_source_url(url)
     assert name == "poetry"
+
+
+def test_annotated_tag() -> None:
+    tag = annotated_tag("my-tag")
+    assert tag == b"my-tag^{}"
+
+
+def test_get_remote_url() -> None:
+    repo = MagicMock(spec=Repo)
+    repo.get_config.return_value.get.return_value = (
+        b"https://github.com/python-poetry/poetry.git"
+    )
+
+    assert Git.get_remote_url(repo) == "https://github.com/python-poetry/poetry.git"
+
+
+def test_urlpathjoin() -> None:
```

Comment:
Can you make the test `parametrized` to add an example with a trailing slash, e.g. `"ssh://git@github.com/org/repo/"`, please?

Reply:
I have made the changes. LMK if I missed something or need to make any further changes. @radoering